File: CodeLibrary/TabHost/BottomNavigationView/AndroidJavaDesignBottomNavigationView/app/src/main/java/com/jackshi/androidjavatabhostfordesign/fragment/HomeFragment.java
package com.jackshi.androidjavatabhostfordesign.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.Spinner;
import android.widget.TextView;

import com.jackshi.androidjavatabhostfordesign.R;
import com.jackshi.androidjavatabhostfordesign.utils.MessageEvent;

import org.greenrobot.eventbus.EventBus;

public class HomeFragment extends Fragment implements CompoundButton.OnCheckedChangeListener, AdapterView.OnItemSelectedListener {

    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    private String mParam1;
    private String mParam2;
    CheckBox mCheckBox1, mCheckBox2, mCheckBox3;
    Spinner mSpinner;

    public HomeFragment() {
    }

    public static HomeFragment newInstance(String param1, String param2) {
        HomeFragment fragment = new HomeFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_home, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        TextView tv = getView().findViewById(R.id.fragment_home_tv);
        tv.setText(mParam1);

        mCheckBox1 = getView().findViewById(R.id.fragment_home_check1);
        mCheckBox2 = getView().findViewById(R.id.fragment_home_check2);
        mCheckBox3 = getView().findViewById(R.id.fragment_home_check3);
        mSpinner = getView().findViewById(R.id.fragment_home_spinner);
        mSpinner.setOnItemSelectedListener(this);
        mCheckBox1.setOnCheckedChangeListener(this);
        mCheckBox2.setOnCheckedChangeListener(this);
        mCheckBox3.setOnCheckedChangeListener(this);
    }

    @Override
    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
        switch (buttonView.getId()) {
            case R.id.fragment_home_check1:
                if (isChecked) {
                    EventBus.getDefault().post(new MessageEvent(0, true));
                    mCheckBox1.setEnabled(false);
                }
                break;
            case R.id.fragment_home_check2:
                if (isChecked) {
                    EventBus.getDefault().post(new MessageEvent(1, true));
                } else {
                    EventBus.getDefault().post(new MessageEvent(1, false));
                }
                break;
            case R.id.fragment_home_check3:
                if (isChecked) {
                    EventBus.getDefault().post(new MessageEvent(3, true));
                } else {
                    EventBus.getDefault().post(new MessageEvent(3, false));
                }
                break;
            default:
                break;
        }
    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        EventBus.getDefault().post(new MessageEvent(2, position));
        if (position == 0) {
            mCheckBox2.setEnabled(false);
            mCheckBox3.setEnabled(true);
        } else if (position == 1) {
            mCheckBox2.setEnabled(false);
            mCheckBox3.setEnabled(true);
        } else if (position == 2) {
            //文字和图标同时都显示时,移除缩放效果才有意义
            mCheckBox2.setEnabled(true);
            mCheckBox3.setEnabled(false);
        } else {
            mCheckBox2.setEnabled(false);
            mCheckBox3.setEnabled(false);
        }
    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {
    }
}

File: CodeLibrary/TabHost/BottomNavigationView/AndroidXkotlinBottomNavigationView/app/src/main/java/com/jackshi/androidxkotlinbottomnavigationview/fragment/HomeFragment.kt
package com.jackshi.androidxkotlinbottomnavigationview.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.CompoundButton
import androidx.fragment.app.Fragment
import com.jackshi.androidxkotlinbottomnavigationview.R
import com.jackshi.androidxkotlinbottomnavigationview.utils.MessageEvent
import kotlinx.android.synthetic.main.fragment_home.*
import org.greenrobot.eventbus.EventBus

private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

class HomeFragment : Fragment(), CompoundButton.OnCheckedChangeListener,
    AdapterView.OnItemSelectedListener {
    private var param1: String? = null
    private var param2: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getString(ARG_PARAM2)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_home, container, false)
    }

    companion object {
        @JvmStatic
        fun newInstance(param1: String, param2: String) =
            HomeFragment().apply {
                arguments = Bundle().apply {
                    putString(ARG_PARAM1, param1)
                    putString(ARG_PARAM2, param2)
                }
            }
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        fragment_home_tv.text = param1
        fragment_home_check1.setOnCheckedChangeListener(this)
        fragment_home_check2.setOnCheckedChangeListener(this)
        fragment_home_check3.setOnCheckedChangeListener(this)
        fragment_home_spinner.onItemSelectedListener = this
    }

    override fun onCheckedChanged(buttonView: CompoundButton, isChecked: Boolean) {
        when (buttonView.id) {
            R.id.fragment_home_check1 -> if (isChecked) {
                EventBus.getDefault().post(MessageEvent(0, true))
                fragment_home_check1.isEnabled = false
            }
            R.id.fragment_home_check2 -> if (isChecked) {
                EventBus.getDefault().post(MessageEvent(1, true))
            } else {
                EventBus.getDefault().post(MessageEvent(1, false))
            }
            R.id.fragment_home_check3 -> if (isChecked) {
                EventBus.getDefault().post(MessageEvent(3, true))
            } else {
                EventBus.getDefault().post(MessageEvent(3, false))
            }
            else -> {
            }
        }
    }

    override fun onNothingSelected(parent: AdapterView<*>?) {
    }

    override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        EventBus.getDefault().post(MessageEvent(2, position))
        when (position) {
            0 -> {
                fragment_home_check2.isEnabled = false
                fragment_home_check3.isEnabled = true
            }
            1 -> {
                fragment_home_check2.isEnabled = false
                fragment_home_check3.isEnabled = true
            }
            2 -> {
                //文字和图标同时都显示时,移除缩放效果才有意义
                fragment_home_check2.isEnabled = true
                fragment_home_check3.isEnabled = false
            }
            else -> {
                fragment_home_check2.isEnabled = false
                fragment_home_check3.isEnabled = false
            }
        }
    }
}